File: pinyougou-parent/pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import java.util.List;
import java.util.Map;

import com.pinyougou.pojo.TbBrand;

import entity.PageResult;


/**
 * 品牌接口
 * @author Administrator
 *
 */
public interface BrandService {
	//查询所有
	public List<TbBrand> findAll();
	
	/**
	 * 返回分页列表
	 * @param pageNum
	 * @param pageSize
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	
	/**
	 * 新增
	 * @param tbBrand
	 */
	public void add(TbBrand tbBrand);
	
	/**
	 * 根据id查询实体
	 * @param id
	 * @return
	 */
	public TbBrand findOne(Long id);
	
	/**
	 * 修改
	 * @param brand
	 * @return
	 */
	public void update(TbBrand brand);
	
	/**
	 * 删除
	 * @param ids
	 */
	public void delete(Long[] ids);
	
	/**
	 * 返回分页列表
	 * @param pageNum
	 * @param pageSize
	 * @return
	 */
	public PageResult findPage(TbBrand brand,int pageNum,int pageSize);
	
	/**
	 * 下拉列表
	 * @return
	 */
	public List<Map> selectOptionList();
	
	
}

File: pinyougou-parent/pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//定义service
		app.service('brandService',function($http){
			
			//查询所有
			this.findAll=function(){
				return $http.get('../brand/findAll.do');
			}
			
			//分页查询
			this.findPage=function(page,rows){
				return $http.get('../brand/findPage.do?page='+page+'&rows='+rows);
			}
			
			
			//查询一个实体
			this.findOne=function(id){
				return $http.get('../brand/findOne.do?id='+id);
			}
			
			//新增
			this.add=function(entity){
				return $http.post('../brand/add.do',entity);
			}
			
			//修改
			this.update=function(entity){
				return $http.post('../brand/update.do',entity);
			}
			
			//删除
			this.dele=function(ids){
				return $http.post('../brand/delete.do?ids='+ids);
			}
			
			//根据条件查询
			this.search=function(page,rows,searchEntity){
				return $http.post('../brand/search.do?page='+page+'&rows='+rows, searchEntity);
			}
			
			//下拉列表
			this.selectOptionList=function(){
				return $http.get('../brand/selectOptionList.do');
			}
			
			
		});
		